src/main/java/serene/Serene.java

```diff
@@ -0,0 +1,402 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Serene {
+    private static final String PARTITION_LINE = "____________________________________________________________";
+    private static final String INPUT_ERROR_MESSAGE = "Uhm... I'm afraid I don't know what you mean by that :/";
+    private static final String INVALID_NUM_ERROR_MESSAGE = "Please enter a valid task number ;-;";
+    private static final String EMPTY_DESC_ERROR_MESSAGE = "Hey! Don't try to make me record nothing for fun :<";
+    private static final String EMPTY_BY_ERROR_MESSAGE = "No time input? Please remember your /by~";
+    private static final String EMPTY_AT_ERROR_MESSAGE = "No time input? Please remember your /at~";
+    private static final String IO_FAIL_MESSAGE = "I/O failed ;-;";
+    private static final int DONE = -1;
+    private static final int CONTINUE = -2;
+    private static final int DEADLINE = 1;
+    private static final int EVENT = 2;
+    private static final int RESPONSE_INDEX_KEYWORD = 0;
+    private static final int RESPONSE_INDEX_BODY = 1;
+    private static final int TASK_INDEX_DESCRIPTION = 0;
+    private static final int TASK_INDEX_OPTIONS = 1;
+    private static final int ERROR_CODE = -1;
+    private static ArrayList<Task> taskList = new ArrayList<>();
+    private static final int SAVED_INDEX_TYPE = 1;
+    private static final int SAVED_INDEX_IS_DONE = 4;
+    private static final int SAVED_INDEX_DESCRIPTION = 7;
+    private static final int TIME_OFFSET = 6;
+    private static int taskCount = 0;
+    private static int statusOfSerene = CONTINUE;
+    private static final String SAVE_FILE_PATH = "data/serene.txt";
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        try {
+            initiateSerene();
+        } catch (IOException e) {
+            System.out.println(IO_FAIL_MESSAGE);
+            printExitMessage();
+            return;
+        }
+        printWelcomeMessage();
+        operateSerene(in);
+        printExitMessage();
+    }
+
+    private static void initiateSerene() throws IOException {
+        File dataDirectory = new File("data");
+        if (!dataDirectory.exists()) {
+            dataDirectory.mkdir();
+            System.out.println("Data directory created~");
+        }
+        File save = new File(SAVE_FILE_PATH);
+        if (save.createNewFile()) {
+            System.out.println("Save file created~");
+            return;
+        }
+        try {
+            readSavedContents(save);
+        } catch (FileNotFoundException e) {
+            System.out.println(IO_FAIL_MESSAGE);
+        }
+    }
+
+    private static void readSavedContents(File save) throws FileNotFoundException {
+        Scanner s = new Scanner(save);
+        while(s.hasNext()) {
+            recoverTask(s.nextLine());
+            taskCount++;
+        }
+    }
+
+    private static void recoverTask(String savedTask) {
```

Comment:
There little or no [javadoc comments](https://se-education.org/guides/conventions/java/basic.html#comments) in your files. You could include them to summarize what the methods are doing.